flocker/acceptance/test_ports.py

```diff
@@ -54,7 +54,8 @@ def setUp(self, cluster):
             },
         }
 
-        cluster.flocker_deploy(self, port_deployment, port_application)
+        self.deployed = cluster.flocker_deploy(
+            self, port_deployment, port_application)
 
```

Comment:
Instead of storing the Deferred, you can just return it from `setUp`. Trial also waits for Deferreds from `setUp`.